### ISSUE TITLE
Make the pedia links to buildings and techs filter for relevance

### ISSUE THREAD

**Issue by magnate** [freeorion/freeorion#3579]
Feature request
===============

Idea
----
The pedia helpfully explains, in many of its articles, that the meter in question is affected by Building Types or Technologies (and also in some cases Policies). It offers hyperlinks to lists of those things. But there is no way of telling which of the things on the list affects the thing you were reading up on. It would be great if the list was filtered to show you only the things which affect the thing you were reading about. 

Gain
----
As the game gets more complex, with a huge proliferation of buildings, techs and policies, it becomes more important to be able to filter the lists of them. Interactive filtering is presumably harder than this suggestion, which could be a stopgap until it is implemented. 

Discussion
----------
If buildings, techs and policies already have flags for what they affect then this should be simple. If they don't, I'd be happy to add them to the data files if someone wanted to code the filtering.

**Comment by geoffthemedio** [comment]
Could you give a specific example of the problem and how you'd like it fixed?

**Comment by magnate** [comment]
Sure, I'm sorry for any lack of clarity. 

1. Open the Pedia
2. Click on Meter Type
3. Click on Stability
4. See where it says "Its value is connected to Building Types" and click on Building Type
5. See the full list of all buildings in the game

At step 5, I'd like to see a list of all buildings which affect Stability, and no others.

**Comment by geoffthemedio** [comment]
How do you define "affect Stability"?

I feel like doing this robustly would involve a way to script a filter with which the UI would generate the list dynamically, rather than requiring someone to manually label a bunch of buildings in a hard-to-maintain way, or manually write a pedia article that contains the list in a slightly less hard-to-maintain way.

**Comment by geoffthemedio** [comment]
Alternatively, meter articles could have programmatically generated lists of content, where the listed stuff has some hard-coded property. This is already done in various cases like lists of species that like something or list of ships of a particular design. In the case of "stability", I'm not sure what you want, but it could be lists of content that has an effect that modifies stability, though this will be of questionable use if it includes any effect that affects stability in any case, as many of those effects are specific to a small number of species, due to likes/dislikes, or other conditions.

**Comment by magnate** [comment]
Yes, the definition of "affect Stability" would have to be specific to the player's race, the whole point of this suggestion is to provide easy access to the likes and dislikes of the race. So yes I think I agree with you that it should be generated dynamically (I'm not sure how you distinguish between dynamically in your first comment and programmatically in your second). 

I'm not too familiar with the stability/government mechanics yet - other than racial likes and dislikes (of buildings, policies etc.), are there other impacts on stability that differ by race? Would the list be simply all the neutral stability-affecting buildings (or policies etc.) - those whose effects are not dependent on race - plus all the liked/disliked ones?

**Comment by geoffthemedio** [comment]
Species can have different baseline stability levels. Any arbitrary game content can also affect any meter via scripted effects.

**Comment by magnate** [comment]
I'm not sure whether that's a reason not to do this? Species have different baseline research and production levels too, but it's still helpful to see what affects RP and PP. I don't know about arbitrary game content - I think perhaps the list won't be exhaustive in that case, but still useful.

**Comment by geoffthemedio** [comment]
I was just replying to your questions...

**Comment by magnate** [comment]
Doh, sorry, I lost the context. Thank you.

**Comment by st-pa-fo** [comment]
well i like the idea too, let me have a look into it.

**Comment by st-pa-fo** [comment]
having looked at the source code, i have discovered that i have to spend more time looking at the source code. in a first step, i would remove all colony 'buildings' of specific species from the list since that is very redundent and makes the list much more readable. see before/after screenshots.

![before--with-species](https://user-images.githubusercontent.com/37298400/148590479-a2042fa1-8cc5-4bad-9a86-deb47884ac8e.png)
![after--without-species](https://user-images.githubusercontent.com/37298400/148590498-72c103e8-e2cd-4339-b345-7d9d39d2c84a.png)

as for more intelligent filtering, i'll have to read up on that ..
but those dozens of colony buildings with all the exact same entries except that the species name is inserted? i think that can go either completely away or maybe move to the Colony Base pedia page
